### DISCUSSION TITLE
Renovate requires app write access for code?

### DISCUSSION THREAD

**Discussion by cpu** [renovatebot/renovate#29362] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi there :wave:

I work on a security sensitive public open source Rust repository. We're looking at adopting Renovate to replace Dependabot, but it appears like installing the Renovate GitHub application requires **write** access to **code**.

![renovate setup](https://github.com/renovatebot/renovate/assets/292650/295a9db3-e787-462b-9c71-8d729f43558d)

Can someone help me understand why this is required? In our ideal deployment model Renovate would only open pull requests and wouldn't require write access to code.

Thanks!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Write access is needed to create branches, branches are needed to create Pull Requests.

My primary advice is that GitHub's branch protections are excellent and if used correctly it means Renovate having write access should be no threat (it does not need write access to main unless you want to automerge).

If you have just a general fear of write access or an inability to use branch protections then there's a workaround - to use a sister app called "Forking Renovate" which does some tricks in order to be able to submit PRs from forks for public repositories

**Reply by HonkingGoose** [reply]
@cpu please also read the [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/) to learn about this topic in more detail.

**Comment by cpu** [comment]
Thank you both for the quick replies. Much appreciated :bow:

> Write access is needed to create branches, branches are needed to create Pull Requests.

Aha, that does make sense.

> If you have just a general fear of write access or an inability to use branch protections then there's a workaround - to use a sister app called "Forking Renovate" which does some tricks in order to be able to submit PRs from forks for public repositories

That sounds like an interesting option. I'll check it out.

> please also read the [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/) to learn about this topic in more detail.

Thank you for the pointer!
